engine/src/cubos/engine/input/plugin.cpp

```diff
@@ -0,0 +1,40 @@
+/// @file
+/// @brief Input plugin entrypoint, registers the input bindings bridge.
+
+#include <cubos/engine/assets/bridges/json.hpp>
+#include <cubos/engine/assets/plugin.hpp>
+#include <cubos/engine/input/bindings.hpp>
+#include <cubos/engine/input/input.hpp>
+#include <cubos/engine/input/plugin.hpp>
+#include <cubos/engine/window/plugin.hpp>
+
+using cubos::core::ecs::EventReader;
+using cubos::core::ecs::Write;
+using cubos::core::io::WindowEvent;
+using namespace cubos::engine;
+
+static void bridge(Write<Assets> assets)
+{
+    assets->registerBridge(".bind", std::make_unique<JSONBridge<InputBindings>>());
+}
+
+static void update(Write<Input> input, EventReader<WindowEvent> events)
+{
+    for (auto event : events)
+    {
+        std::visit([&input](auto e) { input->handle(e); }, event);
```

Comment:
'Catch' the specific event types with the visitor, and add an `auto` catch all for other event types which doesn't call the input object.

Reply:
that requires an overload template, let's not please

Reply:
Sure...